### DISCUSSION TITLE
Audio dropouts when instrument automation enabled in cubase 11

### DISCUSSION THREAD

**Discussion by stuart-beattie** [apohl79/audiogridder#749] [Help]
Further testing in cubase 11. Everything is working really well except automation. I've previously reported automation not being recorded via the plugin UI using cubase, however there's also an issue with audio dropouts occurring as soon as reading automation is activated. When automation is turned off the dropouts stop.

**Comment by apohl79** [comment]
If you can reproduce the drop outs, could you provide the plugin&server logs? Thx!

**Comment by stuart-beattie** [comment]
[AudioGridderPlugin_2021-10-08_21-50-50.log](https://github.com/apohl79/audiogridder/files/7314120/AudioGridderPlugin_2021-10-08_21-50-50.log
[AudioGridderServer_2021-10-08_21-50-29.log](https://github.com/apohl79/audiogridder/files/7314121/AudioGridderServer_2021-10-08_21-50-29.log)
)
Please find attached. The log was recorded creating a new empty project in cubase, adding an instance of audiogridder VST2 instrument, loading remote instance of Diva VST2 and recording a small midi clip. The output parameter on Diva was then automated. Audio dropouts heard on playback with automation being read. Thanks.

**Reply by apohl79** [reply]
Can you try to increase the buffer in AG? The plugin complained about not getting audio data fast enough, thats why you are hearing drop outs:

`[21:51:51.704|client|36a60860] warning: instance (Diva(x64)): read queue empty, waiting for data, try increasing the NumberOfBuffers value`

**Reply by stuart-beattie** [reply]
Its not really making a difference (even up at 30 blocks). I'm observing the same with the effect plugin by the way.

**Reply by apohl79** [reply]
Ok will take a closer look.

Do you have the same behaviour with other vendors? Does it work better, when you use VST3?

**Reply by stuart-beattie** [reply]
as soon as i disable read automation it works perfectly, (even at no buffer).

**Reply by stuart-beattie** [reply]
yes exactly the same thing seen for other plugins

**Reply by stuart-beattie** [reply]
have tried VST2 and VST3

**Reply by apohl79** [reply]
Ok, thx, that should help reproducing this. Created an issue for this.

**Reply by stuart-beattie** [reply]
ah, disarming record on the track seems to reduce the dropouts

**Reply by stuart-beattie** [reply]
have duplicated 10 instances of audiogridder running diva at 1 buffer, with short midi clip and output automation. No dropouts until i arm record on one of them.

**Reply by stuart-beattie** [reply]
one more thing however, if i open up a larger cubase template with a few inserts on the signal path, and then try the same, i get dropouts with one instance of audiogridder with automation enabled, whether I or not I have the track record armed. Again as soon as automation switched off on the track, things work without issue. hope this helps. Thanks

**Reply by apohl79** [reply]
thanks!

**Reply by apohl79** [reply]
I can't reproduce this really. I did a little screen recording with a demo of diva. no dropouts or anything with or without AG buffering. the track was always record armed. maybe you can do the same, so I see what I have to do in cubase, but so far I can't see the issue.

https://user-images.githubusercontent.com/1129160/136823018-6c00df75-d592-4bca-b146-c8aebac2cc84.mp4

**Comment by stuart-beattie** [comment]
I've had another look at this. If I switch my ASIO driver to Generic Low Latency ASIO, the issue goes away 🤔 I'm usually running with the RME Fireface USB ASIO driver (RME babyface PRO interface).

**Comment by stuart-beattie** [comment]
The issue also resolves if I raise the Fireface buffer size from 256 to 1024 samples. Would automation be expected to place significant increased demand on buffer size?

**Reply by apohl79** [reply]
No idea, might be cubase specific

**Reply by stuart-beattie** [reply]
Another observation is that the automation works without issue during recording. Its only during playback that the dropouts are heard.

**Reply by stuart-beattie** [reply]
and another, tried the same using Reaper, same issue, albeit not as bad
